pkg/services/ngalert/notifier/channels/opsgenie.go

```diff
@@ -243,9 +253,9 @@ func (on *OpsgenieNotifier) buildOpsgenieMessage(ctx context.Context, alerts mod
 	bodyJSON.Set("tags", tags)
 	bodyJSON.Set("details", details)
 	apiURL = tmpl(on.APIUrl)
-
 	if tmplErr != nil {
-		on.log.Warn("failed to template Opsgenie message", "err", tmplErr.Error())
+		on.log.Warn("failed to template Opsgenie URL", "err", tmplErr.Error(), "fallback", on.APIUrl)
+		apiURL = on.APIUrl
```

Comment:
That's a fair breakdown IMO. For now we can go with how it's done in other notifiers. In the future though, it might make sense to have one more layer of checks here: After template error, check if it's a valid URL. If it is, use it, else fallback to `OpsgenieAlertURL`.